corehq/motech/generic_inbound/utils.py

```diff
@@ -125,3 +126,27 @@ def reprocess_api_request(request_log):
         request_log.response_status = response.status
         request_log.save()
         make_processing_attempt(response, request_log, is_retry=True)
+
+
+def archive_api_request(request_log, user_id):
+    attempts = request_log.processingattempt_set.filter(xform_id__isnull=False)
+    for attempt in attempts:
```

Comment:
Is there a case where there would be more than one form in a `ProcessingAttempt` that needs to be archived? I see that you asked about this in the ticket. Did you get any clarification?